File: patterns/rule-verriegelung/rules-python/rule_verriegelung.py
from datetime import datetime, timedelta

from openhab import rule, logger
from openhab.triggers import ItemStateUpdateTrigger

until = None


@rule(triggers=[ItemStateUpdateTrigger("MyItem")])
class LatchedRule:
    def execute(self, module, input):
        global until
        now = datetime.now().astimezone()
        if until is not None and until > now:
            logger.info("Event ignoriert, gesperrt bis " + str(until))
            return  # Skip event if timer exists

        until = now + timedelta(days=1)
        logger.info("Regelcode ausgefuehrt, gesperrt bis " + str(until))
        # Regelcode ausfuehren

File: patterns/rule-verriegelung/rules-js/rule-verriegelung.js
let until = null;

rules.JSRule({
  name: "Latched Rule",
  triggers: [triggers.ItemStateUpdateTrigger("MyItem")],
  execute: (event) => {
    const now = time.ZonedDateTime.now();
    if (until !== null && until.isAfter(now)) {
      console.log("Event ignoriert, gesperrt bis " + until.toString());
      return; // Skip event if timer exists
    }

    until = now.plusDays(1);
    console.log("Regelcode ausgefuehrt, gesperrt bis " + until.toString());

    // Regelcode ausfuehren
  }
});
